src/CMake/VisItMacros.cmake

```diff
@@ -120,3 +120,196 @@ MACRO(VISIT_PLUGIN_TARGET_RTOD type)
 ENDMACRO(VISIT_PLUGIN_TARGET_RTOD)
 
 
+##############################################################################
```

Comment:
These 5 functions have about 95% of code in common. Can't they be refactored maybe to one function with some additional params or macro'ized?

Reply:
we could, but since I have already tested this I would rather focus on my time on python changes and not stay stuck in cmake land for too much longer

Reply:
Do you think re-factoring to a single func. would reduce maint. burden? If so, I think its worth doing. As things stand now, if something needs to be adjusted in the way these targets work, we have 5 times as much editing and reviewing to do.

Reply:
yes - done the right way it could. Also given we need to special case for windows that tips me over to support the changes.  Let's address as part of another PR when we can attack the windows support as well or?

Reply:
Ok, will assume you will add a ticket, ok?

Reply:
I went ahead and tackled it, and also added the case for windows.

Reply:
❤️